client/boot/index.js

```diff
@@ -217,6 +218,18 @@ function reduxStoreReady( reduxStore ) {
 		renderLayout( reduxStore );
 	}
 
+	const isLegacyRoute = ( path ) => {
+		return ( /.php$/.test( path ) ||
+			/^\/?$/.test( path ) && ! config.isEnabled( 'reader' ) ||
```

Comment:
Although we thankfully have pretty solid precedence rules, I find these situations really confusing. That is, whenever we have a mix of boolean logic in a sequence. If they are all `||` then it's easy to see that only a single one in the mix has to be true; likewise if they are all `&&` then we know they all need to be true.

What about in `a && b || c`? Does _only_ `c` need to be true or does _either_ `b` or `c` need to be true _with_ `a` being true?

Thus, I usually like to clarify by wrapping different boolean expressions in parenthesis to make the order of evaluation explicit. `a && (b || c)` or `(a && b) || c`